src/graphics/vertex-format.js

```diff
@@ -105,117 +105,111 @@ import {
+            } else if (elementDesc.semantic === SEMANTIC_COLOR) {
+                this.hasColor = true;
+            } else if (elementDesc.semantic === SEMANTIC_TANGENT) {
+                this.hasTangents = true;
+            }
         }
 
-        element = {
-            name: elementDesc.semantic,
-            offset: (vertexCount ? offset : (elementDesc.hasOwnProperty('offset') ? elementDesc.offset : offset)),
-            stride: (vertexCount ? elementSize : (elementDesc.hasOwnProperty('stride') ? elementDesc.stride : this.size)),
-            dataType: elementDesc.type,
-            numComponents: elementDesc.components,
-            normalize: (elementDesc.normalize === undefined) ? false : elementDesc.normalize,
-            size: elementSize
-        };
-        this.elements.push(element);
-
         if (vertexCount) {
-            offset += elementSize * vertexCount;
-        } else {
-            offset += Math.ceil(elementSize / 4) * 4;
+            this.verticesByteSize = offset;
         }
 
-        if (elementDesc.semantic === SEMANTIC_TEXCOORD0) {
-            this.hasUv0 = true;
-        } else if (elementDesc.semantic === SEMANTIC_TEXCOORD1) {
-            this.hasUv1 = true;
-        } else if (elementDesc.semantic === SEMANTIC_COLOR) {
-            this.hasColor = true;
-        } else if (elementDesc.semantic === SEMANTIC_TANGENT) {
-            this.hasTangents = true;
-        }
+        this.update();
     }
 
-    if (vertexCount) {
-        this.verticesByteSize = offset;
-    }
+    static init(graphicsDevice) {
+        var formatDesc = [
+            { semantic: SEMANTIC_TEXCOORD2, components: 4, type: TYPE_FLOAT32 },
+            { semantic: SEMANTIC_TEXCOORD3, components: 4, type: TYPE_FLOAT32 },
+            { semantic: SEMANTIC_TEXCOORD4, components: 4, type: TYPE_FLOAT32 },
+            { semantic: SEMANTIC_TEXCOORD5, components: 4, type: TYPE_FLOAT32 }
+        ];
 
-    this.update();
-}
-
-VertexFormat.init = function (graphicsDevice) {
-    var formatDesc = [
-        { semantic: SEMANTIC_TEXCOORD2, components: 4, type: TYPE_FLOAT32 },
-        { semantic: SEMANTIC_TEXCOORD3, components: 4, type: TYPE_FLOAT32 },
-        { semantic: SEMANTIC_TEXCOORD4, components: 4, type: TYPE_FLOAT32 },
-        { semantic: SEMANTIC_TEXCOORD5, components: 4, type: TYPE_FLOAT32 }
-    ];
-
-    this._defaultInstancingFormat = new VertexFormat(graphicsDevice, formatDesc);
-};
-
-/**
- * @field
- * @static
- * @readonly
- * @name pc.VertexFormat.defaultInstancingFormat
- * @type {pc.VertexFormat}
- * @description Returns {@link pc.VertexFormat} used to store matrices of type {@link pc.Mat4} for hardware instancing.
- */
-Object.defineProperty(VertexFormat, 'defaultInstancingFormat', {
-    get: (function () {
-        return function () {
-            return this._defaultInstancingFormat;
-        };
-    }())
-});
+        this._defaultInstancingFormat = new VertexFormat(graphicsDevice, formatDesc);
```

Comment:
I'd expect _defaultInstancingFormat to be static as well .. not sure how come it wasn't before and how I got this to work. I'm surprised that static init function call access this? I'll have to play with this, but feels wrong (starting with my original code).

Reply:
I've revised the code.

Reply:
That looks great now. I have no idea how my original code worked.

Reply:
Cool. Happy for me to merge then?